File: sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        vector<int> freq (3);
        int idx = 0;

        for (int n : nums) {
            freq[n]++;
        } //for

        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < freq[i]; j++) {
                nums[idx++] = i;
            } //for
        } //for
    } //sortColors
};

File: LeetCode/75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int [] freq = new int [3];
        int ni = 0;

        for (int i = 0; i < nums.length; i++) {
            freq[nums[i]]++;
        } //for

        for (int i = 0; i < freq.length; i++) {
            for (int j = 0; j < freq[i]; j++) {
                nums[ni++] = i;
            } //for
        } //for

    } // sortColors
} // Sol